hardware_integration/src/socket_can_interface.cpp

```diff
@@ -216,4 +217,29 @@ namespace isobus
 		}
 		return retVal;
 	}
+
+	std::vector<std::string> SocketCANInterface::get_CAN_interfaces_list()
+	{
+		struct ifaddrs *ifaddr, *ifa;
+		std::vector<std::string> canInterfaceList {};
+		
+		if (getifaddrs(&ifaddr) == -1) {
+			perror("getifaddrs"); 
+			isobus::CANStackLogger::CAN_stack_log(isobus::CANStackLogger::LoggingLevel::Critical, "[SocketCAN] " + "enable to retrieve the all available CAN Interfaces");
+			return canInterfaceList;
+		}
+
+		for (ifa = ifaddr; ifa != nullptr; ifa = ifa->ifa_next) {
+		
+			if (ifa->ifa_addr != nullptr && ifa->ifa_addr->sa_family == AF_INET) {
+		
+				struct sockaddr_in *addr = reinterpret_cast<struct sockaddr_in*>(ifa->ifa_addr);
+				canInterfaceList.push_back(ifa->ifa_name);
+				
+			}
+		}
+		
+		freeifaddrs(ifaddr);
+		return canInterfaceList;
```

Comment:
Yeah I see, it is much harder than I thought it would be. At this point I'm kind of reconsidering my issue, I feel like it might be better to add it to the stack, but instead let it be on the application side. Optionally providing common interaction functions with socketcan inside another repository, considering `libsocketcan-dev` seems pretty stable. Not sure...

Reply:
Yeah, I think I agree it might be best to leave it "as an exercise to the reader" as they say in academia haha and not include it in here at least. Since it's your issue, I'll leave it to you if you want to close it